src/psyclone/tests/gocean1p0_opencl_test.py

```diff
@@ -196,63 +218,136 @@ def test_invoke_opencl_kernel_call(kernel_outputdir, monkeypatch, debug_mode):
+cl_true,0_8,size_in_bytes,c_loc(field%grid%area_v),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer area_v_device"', ierr)
+      ierr = clenqueuewritebuffer(cmd_queues(1),field%grid%dx_u_device,\
+cl_true,0_8,size_in_bytes,c_loc(field%grid%dx_u),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer dx_u_device"', ierr)
+      ierr = clenqueuewritebuffer(cmd_queues(1),field%grid%dx_v_device,\
+cl_true,0_8,size_in_bytes,c_loc(field%grid%dx_v),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer dx_v_device"', ierr)
+      ierr = clenqueuewritebuffer(cmd_queues(1),field%grid%dx_t_device,\
+cl_true,0_8,size_in_bytes,c_loc(field%grid%dx_t),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer dx_t_device"', ierr)
+      ierr = clenqueuewritebuffer(cmd_queues(1),field%grid%dy_u_device,\
+cl_true,0_8,size_in_bytes,c_loc(field%grid%dy_u),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer dy_u_device"', ierr)
+      ierr = clenqueuewritebuffer(cmd_queues(1),field%grid%dy_v_device,\
+cl_true,0_8,size_in_bytes,c_loc(field%grid%dy_v),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer dy_v_device"', ierr)
+      ierr = clenqueuewritebuffer(cmd_queues(1),field%grid%dy_t_device,\
+cl_true,0_8,size_in_bytes,c_loc(field%grid%dy_t),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer dy_t_device"', ierr)
+      ierr = clenqueuewritebuffer(cmd_queues(1),field%grid%gphiu_device,\
+cl_true,0_8,size_in_bytes,c_loc(field%grid%gphiu),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer gphiu_device"', ierr)
+      ierr = clenqueuewritebuffer(cmd_queues(1),field%grid%gphiv_device,\
+cl_true,0_8,size_in_bytes,c_loc(field%grid%gphiv),0,c_null_ptr,c_null_ptr)
+      call check_status('"clenqueuewritebuffer gphiv_device"', ierr)
+
+    end subroutine write_grid_buffers'''
+    assert expected in generated_code
 
+    # Check that during the first time set-up the previous routines are called
+    # for a kernel which contains a grid property access.
+    expected = '''
+      if (first_time) then
+        first_time = .false.
+        ! ensure opencl run-time is initialised for this psy-layer module
+        call psy_init
+        num_cmd_queues = get_num_cmd_queues()
+        cmd_queues => get_cmd_queues()
+        kernel_compute_kernel_code = get_kernel_by_name("compute_kernel_code")
+        call initialise_device_buffer(out_fld)
+        call initialise_device_buffer(in_out_fld)
+        call initialise_device_buffer(in_fld)
+        call initialise_device_buffer(dx)
+        call initialise_grid_device_buffers(in_fld)
+        xstart = out_fld%internal%xstart
+        xstop = out_fld%internal%xstop
+        ystart = out_fld%internal%ystart
+        ystop = out_fld%internal%ystop
+      out_fld_cl_mem = transfer(out_fld%device_ptr, out_fld_cl_mem)
```

Comment:
Indentation not quite right.

Reply:
See previous indentation comment. I am a bit confused by these.